crates/rethnet_evm/src/block/remote.rs

```diff
@@ -0,0 +1,118 @@
+use std::sync::Arc;
+
+use rethnet_eth::{
+    block::Header,
+    remote::eth::{self, TransactionConversionError},
+    transaction::SignedTransaction,
+    withdrawal::Withdrawal,
+    Address, B256, B64,
+};
+
+use crate::{blockchain::BlockchainError, Block, SyncBlock};
+
+/// A remote block, which lazily loads additional information when requested.
```

Comment:
I found this a bit confusing. As far as I can tell the lazy loading is happening in `RemoteBlockchain::receipt_by_transaction_hash`?

Reply:
Resolved by https://github.com/NomicFoundation/hardhat/pull/4379/commits/3ccbd57e6a4a6c166e588dad85ef7a59fb8bbbb1